spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindableProxyFactory.java

```diff
@@ -250,9 +95,14 @@ public synchronized Object invoke(MethodInvocation invocation) throws Throwable
 	}
 
 	@Override
+	public void afterPropertiesSet() throws Exception {
+		Assert.notNull(this.channelFactory, "ChannelFactory must not be null.");
```

Comment:
The `channelFactory` bean always exists as this gets created as part of `ChannelBindingServiceConfiguration`.